cmake/SetupClangTidy.cmake

```diff
@@ -18,13 +18,17 @@ endif()
 # Remove the checks because:
 # -llvm-header-guard: We use pragma once instead of include guards
 # -google-runtime-int: specifying int32_t and int64_t instead of just int
+# -google-runtime-references: we use return by reference for performance reasons
```

Comment:
I'm not advocating we change our policy, but I'd like to point out that the Google style is not that all functions must return everything in their return value, but that any "return arguments" must be pointers (so it is clear at the call site whether an argument can be modified).  So "performance reasons" is not really a valid rationale for disabling this: it's just a style choice.